model/lib/extensions/presentation_email_extension.dart

```diff
@@ -217,11 +217,14 @@ extension PresentationEmailExtension on PresentationEmail {
       ..searchSnippetPreview = searchSnippetPreview;
   }
 
-  PresentationEmail updateKeywords(Map<KeyWordIdentifier, bool>? newKeywords) {
+  PresentationEmail updateKeywords(Map<KeyWordIdentifier, bool> newKeywords) {
+    final combinedMap = {...(keywords ?? {}), ...newKeywords};
+    combinedMap.removeWhere((key, value) => !value);
```

Comment:
I still not really understand about this

Reply:
Because we update the email value on memory, not the value returned by the server in some cases of confirming invitations from caledar events. We have set the `keywords` values ​​of the previous email to `false` to update the new value to `true`. But those `false` values ​​still exist in the `keywords` of the email. So we need to delete it, to avoid the case when we perform actions with emails such as: Forward, Reply,.. the old `keywords` containing the value false will be set back to the server. Because we are not sure whether the server deletes the keywords when receiving the value false or not. Currently we are only sure they delete when receiving the value null

Reply:
> Currently we are only sure they delete when receiving the value null

Yes that's the spec.